packages/chain-adapters/src/evm/bnbsmartchain/BscChainAdapter.test.ts

```diff
@@ -0,0 +1,633 @@
+      const adapter = new bsc.ChainAdapter(makeChainAdapterArgs())
+      const wallet = await getWallet()
+
+      wallet.ethSendTx = async () => ({
+        hash: '0xe670ec64341771606e55d6b4ca35a1a6b75ee3d5145a99d05921026d1527331',
+      })
+
+      const tx = { wallet, txToSign: {} } as unknown as SignTxInput<ETHSignTx>
+
+      await expect(adapter.signAndBroadcastTransaction(tx)).resolves.toEqual(
+        '0xe670ec64341771606e55d6b4ca35a1a6b75ee3d5145a99d05921026d1527331',
+      )
+    })
+  })
+
+  describe('signMessage', () => {
+    it('should sign a properly formatted signMessageInput object', async () => {
+      const adapter = new bsc.ChainAdapter(makeChainAdapterArgs())
+      const wallet = await getWallet()
+
+      const message: SignMessageInput<ETHSignMessage> = {
+        wallet,
+        messageToSign: {
+          message: 'Hello world 111',
+          addressNList: toAddressNList(adapter.getBIP44Params({ accountNumber: 0 })),
+        },
+      }
+
+      await expect(adapter.signMessage(message)).resolves.toEqual(
+        '0x05a0edb4b98fe6b6ed270bf55aef84ddcb641512e19e340bf9eed3427854a7a4734fe45551dc24f1843cf2c823a73aa2454e3785eb15120573c522cc114e472d1c',
+      )
+    })
+
+    it('should throw if wallet.ethSignMessage returns null', async () => {
+      const adapter = new bsc.ChainAdapter(makeChainAdapterArgs())
+      const wallet = await getWallet()
+
+      wallet.ethSignMessage = async () => null
+
+      const message: SignMessageInput<ETHSignMessage> = {
+        wallet,
+        messageToSign: {
+          message: 'Hello world 111',
+          addressNList: toAddressNList(adapter.getBIP44Params({ accountNumber: 0 })),
+        },
+      }
+
+      await expect(adapter.signMessage(message)).rejects.toThrow(
+        /EvmBaseAdapter: error signing message/,
+      )
+    })
+  })
+
+  describe('broadcastTransaction', () => {
+    it('should correctly call sendTx and return its response', async () => {
+      const expectedResult = 'success'
+
+      const httpProvider = {
+        sendTx: jest.fn().mockResolvedValue(expectedResult),
+      } as unknown as unchained.bnbsmartchain.V1Api
+
+      const args = makeChainAdapterArgs({ providers: { http: httpProvider } })
+      const adapter = new bsc.ChainAdapter(args)
+
+      const mockTx = '0x123'
+      const result = await adapter.broadcastTransaction(mockTx)
+
+      expect(args.providers.http.sendTx).toHaveBeenCalledWith<any>({ sendTxBody: { hex: mockTx } })
+      expect(result).toEqual(expectedResult)
+    })
+  })
+
+  describe('buildSendTransaction', () => {
+    const accountNumber = 0
+
+    it('should throw if passed tx has no "to" property', async () => {
+      const adapter = new bsc.ChainAdapter(makeChainAdapterArgs())
+
+      const tx = {
+        wallet: await getWallet(),
+        accountNumber,
+        value,
+        chainSpecific: makeChainSpecific({ erc20ContractAddress: contractAddress }),
```

Comment:
Kind of a weird one now that we have `bep20` namespace

Reply:
https://github.com/shapeshift/lib/pull/1197/commits/6344309f441f71641c53a4eda31586cc3e352db1